### ISSUE TITLE
Web Accessibility (or UX) banch

### ISSUE THREAD

**Issue by thiagotpc** [kamranahmedse/developer-roadmap#33]
Nice job! 

I would suggest creating a branch to include "Web Accessibility" or "User Experience" on the Front-End roadmap. 

If opt for Web Accessibility, the topics would be:
- Use cases and rules;
- How to use screenreaders,
- WCAG;
- ARIA;

If opt for User Experience, the topics would be:
- Principles of Design on Design Layout;
- Navigation Design;
- Web Usability;
- Web Accessibility;
- User Testings;

I think that Web Accessibility is very important in context of coding web pages and web applications. And this is a responsibility of front-end rule.  

About the User Experience, I think that this additional skill will allow deliver interfaces with most quality.

**Comment by Ref34t** [comment]
I'm agreed with you every developer should know more about user experience design

**Comment by marceloogeda** [comment]
Good thinking! 

I would like to add just more one topic that I believe to be part of the context of Web Accessibility.

- Semantic HTML;

Good semantic HTML also improves the accessibility of web. So I believe that it belongs to the same context.

**Comment by rbenson** [comment]
I was going to add this as an issue.

@thiagotpc 
I recommend keeping a11y and ux separate, they overlap, but once you get down in the weeds they differ. 

> Use cases and rules;

Not sure what you mean here. In terms of rules, do you mean rules in UAT or a11y guidelines? Countries have their own laws, and a lot point to WCAG, but not all. The US has Section 508 and the ADA. 508 finally got approved in January that says follow WCAG 2.0, however currently it has it's own standards, which is a modified version of WCAG 1.0.

>How to use screenreaders,

This should be how to use assistive technology. Screen readers just cover people who are blind, which leaves out people who:

 - are visually impaired  (they can see, but need things blown up to 300%+)
 - have mobility disabilities
 - are color blind  
 - have cognitive or learning disabilities

@marceloogeda I think semantic HTML should be part of learning HTML not web accessibility. If you are using semantic code, it helps in web accessibility, but also helps in SEO, testing, and debugging.

**Comment by damengilland** [comment]
##Section 508 Standards

Is your business part of a government organization, or do you receive money from the government?  If so, you are most likely required to comply with Section 508 accessibility guidelines for users with disabilities.

Read More:

- Section 508 Standards - United States Access Board || https://www.access-board.gov/guidelines-and-standards/communications-and-it/about-the-section-508-standards/section-508-standards#22 
- DHS Section 508 Compliance Test Processes | Homeland Security || https://www.dhs.gov/publication/dhs-section-508-compliance-test-processes 
- A few HTML tips ★ Mozilla Hacks – the Web developer blog || https://hacks.mozilla.org/2016/08/a-few-html-tips/ 

## Color Blind

* Spectrum - Chrome Web Store || https://chrome.google.com/webstore/detail/spectrum/ofclemegkcmilinpcimpjkfhjfgmhieb

**Comment by rbenson** [comment]
@damengilland - it is a little more complicated than that. The federal government follows Section 508. State and local governments follow 504, which allows them to decide what they want to do. It is split not quite 50-50 between the  states, with a few outliers. The states either say do 508, or do WCAG 2.0 AA. The outliers do a hybrid approach, since WCAG doesn't cover things like hardware. 

Further, you linked to 1194.22, which is  fine, however a lot of the bigger agencies/departments (SSA, DHS, HHS, etc) do not just use 1194.22. They use 1194.22 when it is a website, 1194.21 when a scripting language is used to create the UI. 1194.31 is used by default when any of the 1194.2x are used, because .2x ask 'did you code it correctly?', 1194.31 says 'can a person with a disability actually use the thing - given that it is coded correctly?" 1194.41 is also used by default, if a .2x is used. 

There are similar laws in other countries too, most point to WCAG 2.0 AA

**Comment by FadiZahhar** [comment]
How I can contribute on this ? to add value interesting.

**Comment by damengilland** [comment]
@rbenson, totally agree.  It's all part of the path.  I'd say the biggest issue I see, working with agencies, is a lack of basic understanding about accessibility to start with.  Someone does some work for a state government entity, or a non-profit funded by one, and just looks at the site and calls it "done" without knowing about various rules that they are bound by.

Having a path that explores these accessibility and compliance guidelines... certainly important.  I don't know the best place to start... just raising awareness for developers though probably.  "Legal Coding 101" -- ha!  There's all sorts of accessibility, COPPA, CAN SPAM, EU Cookie Policy stuff to learn... not to mention various TOS for sending emails while using Mail Chimp / Mailgun / Twillio / etc.  It isn't enough to just make things "work" -- there's a lot to this.

**Comment by togiberlin** [comment]
It doesn't cover web accessibility, but UX...
https://github.com/togiberlin/ui-ux-designer-roadmap

**Comment by jpdevries** [comment]
'Web' is short for World-Wide Web, and world-wide is about as inclusive as it gets, so you can't become a Web Developer without inclusive design patterns and incorporating accessibility into your design process.

I think the good news is inclusive design patterns are a grand slam. You get accessibility, performance (as in reliability), speed, and SEO all from the same patterns. Perhaps there is a way to illustrate that in the roadmap.
